### PR TITLE
POL-317_Updated cost details

### PR DESCRIPTION
**Pull request by chandravurubindi**
Added code to fetch cost and change the escalation label for consistency.

-Updated the policy to fetch and display the cost for each resource under all the subscription ids from Optima

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/old_snapshots/README.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 This Policy finds Azure snapshots older than the specified days and deletes them.
 
+### Policy savings details
+
+The policy includes the estimated savings. The estimated savings is recognized if the resource is terminated. Optima is used to receive the estimated savings which is the product of the most recent full day's cost of the resource * 30. The savings is displayed in the Estimated Monthly Savings column. If the resource can not be found in Optima the value is N/A. The incident message detail includes the sum of each resource Estimated Monthly Savings as Total Estimated Monthly Savings.
+If the user is missing the minimum required role of `billing_center_viewer` or if there is no enough data received from Optima to calculate savings, appropriate message is displayed in the incident detail message along with the estimated monthly savings column value as N/A in the incident table.
```
Comment (douglaswth) [inline cost/azure/old_snapshots/README.md:10]:
```suggestion
If the user is missing the minimum required role of `billing_center_viewer` or if there is not enough data received from Optima to calculate savings, an appropriate message is displayed in the incident detail message along with the estimated monthly savings column value as N/A in the incident table.
```

File: cost/azure/old_snapshots/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## v2.4
+
+- Updated policy to fetch cost details for multiple subscription ids from Optima
```
Comment (douglaswth) [inline cost/azure/old_snapshots/CHANGELOG.md:5]:
```suggestion
- Updated policy to fetch cost details for multiple subscription IDs from Optima
```

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt [hunk 3]
```diff
@@ -115,14 +117,144 @@ datasource "ds_azure_get_snapshots" do
...
 # Scripts
 ###############################################################################
 
+script "js_top_level_bc", type: "javascript" do
+  parameters "billing_centers"
+  result "filtered_billing_centers"
+  code <<-EOS
+  var filtered_billing_centers =
+    _.reject(billing_centers, function(bc){ return bc.parent_id != null });
+EOS
```
Comment (douglaswth) [inline cost/azure/old_snapshots/azure_delete_old_snapshots.pt:202]:
```suggestion
  EOS
```

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt [hunk 4]
```diff
@@ -156,26 +288,104 @@ script "js_filter_snapshots", type: "javascript" do
             timeCreated:snapshot['timeCreated'],
             subscriptionId:snapshot['subscriptionId'],
             subscriptionName:snapshot['subscriptionName'],
-            age:age
+            age:age,
+			savings:"N/A"
```
Comment (douglaswth) [inline cost/azure/old_snapshots/azure_delete_old_snapshots.pt:292]:
```suggestion
            savings:"N/A"
```

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt [hunk 5]
```diff
@@ -156,26 +288,104 @@ script "js_filter_snapshots", type: "javascript" do
...
 end
 
 policy "pol_azure_delete_old_snapshots" do
-  validate $ds_filter_snapshots do
-    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Old Snapshots Found"
+  validate $ds_snapshot_cost_mapping do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data.instances }} Old Snapshots Found"
+  detail_template <<-EOS
+  {{data.message}}.\n
+  EOS
```
Comment (douglaswth) [inline cost/azure/old_snapshots/azure_delete_old_snapshots.pt:382]:
```suggestion
    detail_template <<-EOS
{{data.message}}.
EOS
```

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt [hunk 6]
```diff
@@ -189,6 +399,9 @@ policy "pol_azure_delete_old_snapshots" do
       field "name" do
         label "Name"
       end
+    field "savings" do
+      label "Estimated Monthly Savings"
+    end
```
Comment (douglaswth) [inline cost/azure/old_snapshots/azure_delete_old_snapshots.pt:404]:
```suggestion
      field "savings" do
        label "Estimated Monthly Savings"
      end
```

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt [hunk 7]
```diff
@@ -156,26 +288,104 @@ script "js_filter_snapshots", type: "javascript" do
...
+  detail_template <<-EOS
+  {{data.message}}.\n
+  EOS
     # Send email report
     escalate $send_email_report
     # Delete snapshot if user approves
     escalate $process_snapshots
-    check eq(size(data), 0)
-    export do
+    check eq(size(val(data, "instances")), 0)
```
Comment (douglaswth) [inline cost/azure/old_snapshots/azure_delete_old_snapshots.pt:387]: You may want to add a `hash_include "instances"` as mentioned in [Describing the Policy Conditions](https://docs.rightscale.com/policies/developers/reference/v20180301/policy_template_language.html#describing-the-policy-conditions) to ensure that new notifications only show up when the `instances` field of the `data` changes if there are other items in `data` which may changes.

**Review by douglaswth — CHANGES_REQUESTED** [review]

File: cost/azure/old_snapshots/azure_delete_old_snapshots.pt (hunk 7 above)
Comment (chandravurubindi) [reply cost/azure/old_snapshots/azure_delete_old_snapshots.pt:387]: Thanks for the suggestion!
Added hash_include "instances", currently I am testing this change, and it's in progress, I will puh this change to GitHub once it's working fine as expected!

Comment (chandravurubindi) [reply cost/azure/old_snapshots/azure_delete_old_snapshots.pt]: Tested hash_include changes, its working as expected, hence pushed this change to GitHub.  Pls review it.

**Review by nshivappa — APPROVED** [review]
LGTM

**Review by douglaswth — APPROVED** [review]

**Pull request merged**